### ISSUE TITLE
documentation - The testing guide doesn't show how to use lcov/pycoverage for yourself (Trac #1582)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1574]
add that info to

http://software.icecube.wisc.edu/icerec_trunk/complete_testing_guide.html

Migrated from https://code.icecube.wisc.edu/ticket/1582
```json
{
    "status": "closed", 
    "changetime": "2019-01-12T00:11:31", 
    "description": "add that info to\n\nhttp://software.icecube.wisc.edu/icerec_trunk/complete_testing_guide.html", 
    "reporter": "nega", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1547251891330467", 
    "component": "cmake", 
    "summary": "documentation - The testing guide doesn't show how to use lcov/pycoverage for yourself", 
    "priority": "normal", 
    "keywords": "documentation", 
    "time": "2016-03-08T01:25:15", 
    "milestone": "", 
    "owner": "nega", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160308T01:29:20: nega commented:

> mrichman reports using:
> 
> ```#!bash
> 
> #!/bin/bash
> if [ ! -d ../output ]; then
>     mkdir ../output
> fi
> PROJ=pybdt
> lcov -d $PROJ -z
> lcov -d $PROJ -c -i -o test_base.info
> ./env-shell.sh ctest -R $PROJ
> lcov -d $PROJ -c -o test.info
> lcov -r test.info '/usr/include/*' '$ENV{I3_PORTS}/*' -o reports.info
> genhtml -o ../output`date +%Y-%m-%d` reports.info
> ```

**Comment by icecube-trac** [comment]
Trac update at 20160308T01:36:51: richman commented:

> Yeah, that seems to work for me.  Could probably be generalized to take `$PROJ` as an argument, but otherwise default to the whole tree.  Maybe output should go to a subdir named after `$PROJ`.
> 
> Also, one should set `genhtml_branch_coverage = 1` and `lcov_branch_coverage = 1` in `~/.lcovrc` to get branch coverage data.
> 
> Maybe it would be overkill, but I could imagine wrapping this into per-project and full-metaproject `make` targets.

**Comment by icecube-trac** [comment]
Trac update at 20160308T18:08:35: nega commented:

> lcov documentation in r2453/IceTray

**Comment by icecube-trac** [comment]
Trac update at 20160308T19:37:46: 
* nega commented:

> In changeset 2457:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2457"
> add python coverage docs
> 
> closes #1574
> ```


* nega changed status from "new" to "closed"
* nega changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190112T00:11:31: nwhitehorn commented:

> In changeset 170210:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170210"
> add python coverage docs
> 
> closes #1574
> ```
